cmd/sonobuoy/app/gen_test.go

```diff
@@ -244,44 +257,105 @@ func TestResolveConfig(t *testing.T) {
+				if err != nil {
+					t.Fatalf("Failed to create temp file for test: %v", err)
+				}
+				defer os.Remove(tmpFile.Name())
+				err = ioutil.WriteFile(tmpFile.Name(), []byte(tc.configFileContents), 0666)
+				if err != nil {
+					t.Fatalf("Failed to write temp config file for test: %v", err)
 				}
+				tc.input = strings.Replace(tc.input, dynamicConfigFileName, tmpFile.Name(), -1)
 			}
 
-			conf := tc.input.resolveConfig()
+			// Simulate parsing of input via CLI.
+			gflagset := &genFlags{}
+			fs := GenFlagSet(gflagset, EnabledRBACMode)
+			if err := fs.Parse(strings.Split(tc.input, " ")); err != nil {
+				t.Fatalf("Failed to parse CLI input %q: %v", tc.input, err)
+			}
+			genConfig, err := gflagset.Config()
+			if err != nil {
+				t.Fatalf("Failed to generate GenConfig: %v", err)
+			}
+
+			if genConfig.Config.Namespace != tc.expected.Namespace {
+				t.Errorf("Expected namespace %v but got %v", tc.expected.Namespace, genConfig.Config.Namespace)
+			}
+
+			if genConfig.Config.WorkerImage != tc.expected.WorkerImage {
+				t.Errorf("Expected worker image %v but got %v", tc.expected.WorkerImage, genConfig.Config.WorkerImage)
+			}
 
-			if conf.Namespace != tc.expected.Namespace {
-				t.Errorf("Expected namespace %v but got %v", tc.expected.Namespace, conf.Namespace)
+			if genConfig.Config.ImagePullPolicy != tc.expected.ImagePullPolicy {
+				t.Errorf("Expected image pull policy %v but got %v", tc.expected.ImagePullPolicy, genConfig.Config.ImagePullPolicy)
 			}
 
-			if conf.WorkerImage != tc.expected.WorkerImage {
-				t.Errorf("Expected worker image %v but got %v", tc.expected.WorkerImage, conf.WorkerImage)
+			if genConfig.Config.Aggregation.TimeoutSeconds != tc.expected.Aggregation.TimeoutSeconds {
+				t.Errorf("Expected timeout %v but got %v", tc.expected.Aggregation.TimeoutSeconds, genConfig.Config.Aggregation.TimeoutSeconds)
 			}
 
-			if conf.ImagePullPolicy != tc.expected.ImagePullPolicy {
-				t.Errorf("Expected image pull policy %v but got %v", tc.expected.ImagePullPolicy, conf.ImagePullPolicy)
+			if !reflect.DeepEqual(genConfig.Config.PluginSelections, tc.expected.PluginSelections) {
+				t.Errorf("expected PluginSelections %v but got %v", tc.expected.PluginSelections, genConfig.Config.PluginSelections)
 			}
 
-			if conf.Aggregation.TimeoutSeconds != tc.expected.Aggregation.TimeoutSeconds {
-				t.Errorf("Expected timeout %v but got %v", tc.expected.Aggregation.TimeoutSeconds, conf.Aggregation.TimeoutSeconds)
+			if !reflect.DeepEqual(genConfig.Config.Resources, tc.expected.Resources) {
+				t.Errorf("expected resources %v but got %v", tc.expected.Resources, genConfig.Config.Resources)
 			}
 
-			if !reflect.DeepEqual(conf.PluginSelections, tc.expected.PluginSelections) {
-				t.Errorf("expected PluginSelections %v but got %v", tc.expected.PluginSelections, conf.PluginSelections)
+			if diff := pretty.Compare(genConfig.StaticPlugins, tc.expectedGenConfigPlugins.StaticPlugins); diff != "" {
```

Comment:
Added 2 checks here just for the static/dynamic plugins. Consistent with the existing test which is just checking for key bits of data.

If we wanted to change these tests to check the entire genconfig we should probably just serialize the values and use golden files since the objects are so big.